vuln-reach/src/javascript/lang/accesses.rs

```diff
@@ -165,17 +165,23 @@ impl<'a> AccessGraph<'a> {
     /// NOTE: `test` in this example is not declared as variable and thus
     /// hoisted to the global scope. This is why the accessor of `bar` cannot be
     /// `foo`, otherwise the access from `baz` to `bar` would not be detected.
-    fn find_accessor(cursor_cache: &mut TreeCursorCache<'a>, node: Node<'a>) -> Option<Node<'a>> {
+    fn find_accessor(
+        cursor_cache: &mut TreeCursorCache<'a>,
+        mut node: Node<'a>,
+    ) -> Option<Node<'a>> {
         let cursor = cursor_cache.cursor(node).unwrap();
+        let mut cur_depth = 0;
 
         for (depth, parent) in cursor.parents().enumerate() {
             match parent.kind() {
                 // Check depth to avoid declarations using themselves as accessor.
-                "class_declaration" | "function_declaration" if depth > 0 => {
+                "class_declaration" | "function_declaration" if depth > cur_depth => {
                     return parent.child_by_field_name("name");
                 },
+                // Find the next parent identifier if node is an anonymous declaration.
                 "class" | "function" | "arrow_function" => {
-                    return Self::find_accessor(cursor_cache, parent);
+                    cur_depth = depth;
+                    node = parent;
```

Comment:
I don't like this part in this form because its effects ("restarting" the computation) are not super evident compared to recursing. Happy to accept suggestions here.

Reply:
I'm fine with it tbh. Though I didn't really mind the recursion either (and it's probably *simpler* imo).

Reply:
The recursion is definitely easier to reason about, at least. My concern is more with the fact that this is on a hot path and Rust has no tail recursion and I'm not sure that can be inlined away. Then again I ran the benchmarks and there is so much variance as to make any assessment inconsistent, so it would probably be fine to stick with recursion for clarity's sake.